### PR TITLE
Walletconnect 2.0

### PR DESCRIPTION
**Pull request by 0x-General**

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -24,8 +24,8 @@ utxo:
 types:
 - packages/types/**/*
 
-wc:
-- packages/walletconnect/**/*
```
Comment (chillios-ts) [inline .github/labeler.yml:28]: you should probably leave this one and add `tw` separately

File: README.md [hunk 2]
```diff
@@ -114,7 +114,7 @@ This repo contains packages around SwapKit sdk and its integrations with differe
 | [@thorswap-lib/toolbox-cosmos](./packages/toolbox-cosmos/README.md) | Toolkit to integrate Cosmos chains | THOR, ATOM, BNB                                 |
 | [@thorswap-lib/keystore](./packages/keystore/README.md)             | Keystore implementation            | All chains supported by toolboxes               |
 | [@thorswap-lib/ledger](./packages/ledger/README.md)                 | Ledger implementation              | All chains supported by toolboxes               |
-| [@thorswap-lib/walletconnect](./packages/walletconnect/README.md)   | WalletConnect implementation       | THOR, BNB, ETH                                  |
```
Comment (chillios-ts) [inline README.md:117]: Same as we now hawe 2 packages no?

File: packages/walletconnect/package.json [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
-    "@thorswap-lib/helpers": "workspace:*",
-    "@thorswap-lib/types": "workspace:*",
-    "@walletconnect/client": "1.8.0",
-    "@walletconnect/qrcode-modal": "1.8.0",
-    "@walletconnect/web3-provider": "1.8.0",
-    "base64-js": "1.5.1",
-    "bech32": "2.0.0"
+    "@ethersproject/bytes": "^5.7.0",
+    "@ethersproject/strings": "^5.7.0",
+    "@thorswap-lib/toolbox-evm": "workspace:^",
```
Comment (chillios-ts) [inline packages/walletconnect/package.json:6]: This should be in dev & peer dependencies with being marked as external module in `vite.config.js` as those will be installed with `toolbox-evm`

File: packages/walletconnect/src/helpers.ts [hunk 4]
```diff
@@ -1,70 +1,48 @@
...
+        accounts.find((account) => account.startsWith(chainToChainId(Chain.THORChain))) || '',
+      );
+    case Chain.Ethereum:
+      return getAddressFromAccount(
+        accounts.find((account) => account.startsWith(chainToChainId(Chain.Ethereum))) || '',
+      );
+    default:
+      return getAddressFromAccount(
+        accounts.find((account) => account.startsWith(chainToChainId(Chain.Ethereum))) || '',
+      );
```
Comment (chillios-ts) [inline packages/walletconnect/src/helpers.ts:33]:
```suggestion
	return getAddressFromAccount(
        accounts.find((account) => account.startsWith(chainToChainId(chain))) || '',
      );
```

**Review by chillios-ts — CHANGES_REQUESTED** [review]

File: packages/walletconnect/vite.config.ts [hunk 5]
```diff
@@ -1,25 +1,12 @@
...
-        '@ethersproject/bignumber',
-        '@ethersproject/bytes',
-        '@ethersproject/contracts',
-        '@ethersproject/strings',
-        '@thorswap-lib/toolbox-cosmos',
-        '@thorswap-lib/toolbox-evm',
-        'bech32-buffer',
-        'cosmos-client',
-      ],
+      entry: resolve(__dirname, 'src/index.ts'),
```
Comment (chillios-ts) [inline packages/walletconnect/vite.config.ts:9]: Still missing external modules (peerDependencies) in config

**Review by chillios-ts — CHANGES_REQUESTED** [review]

File: packages/walletconnect/vite.config.ts (hunk 5 above)
Comment (0x-General) [reply packages/walletconnect/vite.config.ts:9]: oh yeah, sorry, forgot about that. Pushed!

**Review by chillios-ts — APPROVED** [review]

**Pull request merged**